static/app/components/customResolutionModal.tsx

```diff
@@ -1,84 +1,83 @@
+  const [version, setVersion] = useState('');
+  const currentUser = configStore.get('user');
 
-class CustomResolutionModal extends Component<Props, State> {
-  state: State = {
-    version: '',
+  const onChange = (value: any) => {
+    setVersion(value.item);
   };
 
-  onChange = (value: string | number | boolean) => {
-    this.setState({version: value as string}); // TODO(ts): Add select value type as generic to select controls
+  const onAsyncFieldResults = (results: Release[]) => {
+    return results.map(release => {
+      const isAuthor = release.authors.some(
+        author => author.email && author.email === currentUser?.email
+      );
+      return {
+        item: release.version,
+        label: <Version version={release.version} anchor={false} />,
+        details: (
+          <span>
+            {t('Created')} <TimeSince date={release.dateCreated} />
+            {isAuthor ? <Fragment> — {t('You committed')}</Fragment> : null}
+          </span>
+        ),
+        release,
+      };
+    });
   };
 
-  onAsyncFieldResults = (results: Release[]) =>
-    results.map(release => ({
-      value: release.version,
-      label: <Version version={release.version} anchor={false} />,
-      details: (
-        <span>
-          {t('Created')} <TimeSince date={release.dateCreated} />
-        </span>
-      ),
-      release,
-    }));
-
-  render() {
-    const {orgSlug, projectSlug, closeModal, onSelected, Header, Body, Footer} =
-      this.props;
-    const url = projectSlug
-      ? `/projects/${orgSlug}/${projectSlug}/releases/`
-      : `/organizations/${orgSlug}/releases/`;
+  const url = props.projectSlug
+    ? `/projects/${props.orgSlug}/${props.projectSlug}/releases/`
+    : `/organizations/${props.orgSlug}/releases/`;
 
-    const onSubmit = (e: React.FormEvent) => {
-      e.preventDefault();
-      onSelected({inRelease: this.state.version});
-      closeModal();
-    };
+  const onSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    props.onSelected({inRelease: version});
+    props.closeModal();
+  };
 
-    return (
-      <form onSubmit={onSubmit}>
-        <Header>{t('Resolved In')}</Header>
-        <Body>
-          <SelectAsyncField
-            label={t('Version')}
-            id="version"
-            name="version"
-            onChange={this.onChange}
-            placeholder={t('e.g. 1.0.4')}
-            url={url}
-            onResults={this.onAsyncFieldResults}
-            onQuery={query => ({query})}
-          />
-        </Body>
-        <Footer>
-          <Button type="button" css={{marginRight: space(1.5)}} onClick={closeModal}>
-            {t('Cancel')}
-          </Button>
-          <Button type="submit" priority="primary">
-            {t('Save Changes')}
-          </Button>
-        </Footer>
-      </form>
-    );
-  }
+  return (
+    <form onSubmit={onSubmit}>
+      <props.Header>{t('Resolved In')}</props.Header>
+      <props.Body>
```

Comment:
Can we spread props instead of using these like this. It's a one-off and a bit of an eyesore.